packages/ngx-widgets/src/app/area-size/area-size.directive.ts

```diff
@@ -0,0 +1,17 @@
+import { Directive, ElementRef, HostListener, Input, Renderer2 } from '@angular/core';
+
+@Directive({
+  selector: '[textAriaResize]',
+})
+export class AreaSizeDirective {
+  private element: HTMLElement = this.elementRef.nativeElement;
+
+  constructor(private elementRef: ElementRef, private renderer: Renderer2) {}
+
+  @HostListener('input', ['$event'])
+  resizeOnInput() {
+    const input = event.target as HTMLInputElement;
+    this.renderer.setStyle(this.element, 'height', '1px');
```

Comment:
Why do you set to `1px` here only to set it to the `scrollHeight` immediately afterwards?

Reply:
When I remove a line in the text area. scrollHeight has the same height and necessary to redefine it.

Reply:
Ok I understand why you're doing it now.
I think you need to ensure as well that `overflow-y` is set to `hidden` to account for the scrollbar appearing for OS / browser settings where a scrollbar consumes space. The scrollbar will flash as well as push the text resulting possibly in an incorrect calculation. Test this on several browsers.

I don't know if this is the best solution as it's doing a re-layout of the component on each input change. I can think of other ways to do this but they are certainly more involved with separate height calculations in another element. So if this works and doesn't give any negative experiences to the user (flashing, etc...), i'm ok with it.